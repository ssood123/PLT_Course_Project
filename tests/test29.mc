function void main()
{
  int matrix[3][3] m1;
  int matrix[3][3] m2;
  int i;
  int j;

  m1 = [[1,2,3],[4,5,6],[7,8,9]];
  m2 = transpose(m1);
  
  for(i = 0; i < lenRow(m2); i = i + 1)
  {
    for (j = 0; j < lenCol(m2); j = j + 1)
    {
      print(m2[i][j]);
    }
  }
  

}
