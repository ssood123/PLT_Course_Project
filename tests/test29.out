; ModuleID = 'integraph'
source_filename = "integraph"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main() {
entry:
  %m1 = alloca [3 x [3 x i32]], align 4
  %m2 = alloca [3 x [3 x i32]], align 4
  %i = alloca i32, align 4
  %j = alloca i32, align 4
  store [3 x [3 x i32]] [[3 x i32] [i32 1, i32 2, i32 3], [3 x i32] [i32 4, i32 5, i32 6], [3 x i32] [i32 7, i32 8, i32 9]], [3 x [3 x i32]]* %m1, align 4
  %tempMatrix = alloca [3 x [3 x i32]], align 4
  %m11 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 0, i32 0
  %m12 = load i32, i32* %m11, align 4
  %tempMatrix3 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 0, i32 0
  store i32 %m12, i32* %tempMatrix3, align 4
  %m14 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 0, i32 1
  %m15 = load i32, i32* %m14, align 4
  %tempMatrix6 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 1, i32 0
  store i32 %m15, i32* %tempMatrix6, align 4
  %m17 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 0, i32 2
  %m18 = load i32, i32* %m17, align 4
  %tempMatrix9 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 2, i32 0
  store i32 %m18, i32* %tempMatrix9, align 4
  %m110 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 1, i32 0
  %m111 = load i32, i32* %m110, align 4
  %tempMatrix12 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 0, i32 1
  store i32 %m111, i32* %tempMatrix12, align 4
  %m113 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 1, i32 1
  %m114 = load i32, i32* %m113, align 4
  %tempMatrix15 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 1, i32 1
  store i32 %m114, i32* %tempMatrix15, align 4
  %m116 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 1, i32 2
  %m117 = load i32, i32* %m116, align 4
  %tempMatrix18 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 2, i32 1
  store i32 %m117, i32* %tempMatrix18, align 4
  %m119 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 2, i32 0
  %m120 = load i32, i32* %m119, align 4
  %tempMatrix21 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 0, i32 2
  store i32 %m120, i32* %tempMatrix21, align 4
  %m122 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 2, i32 1
  %m123 = load i32, i32* %m122, align 4
  %tempMatrix24 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 1, i32 2
  store i32 %m123, i32* %tempMatrix24, align 4
  %m125 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m1, i32 0, i32 2, i32 2
  %m126 = load i32, i32* %m125, align 4
  %tempMatrix27 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0, i32 2, i32 2
  store i32 %m126, i32* %tempMatrix27, align 4
  %tempMatrix28 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix, i32 0
  %tempMatrix29 = load [3 x [3 x i32]], [3 x [3 x i32]]* %tempMatrix28, align 4
  store [3 x [3 x i32]] %tempMatrix29, [3 x [3 x i32]]* %m2, align 4
  store i32 0, i32* %i, align 4
  br label %while

while:                                            ; preds = %merge, %entry
  %i41 = load i32, i32* %i, align 4
  %tmp42 = icmp slt i32 %i41, 3
  br i1 %tmp42, label %while_body, label %merge43

while_body:                                       ; preds = %while
  store i32 0, i32* %j, align 4
  br label %while30

while30:                                          ; preds = %while_body31, %while_body
  %j37 = load i32, i32* %j, align 4
  %tmp38 = icmp slt i32 %j37, 3
  br i1 %tmp38, label %while_body31, label %merge

while_body31:                                     ; preds = %while30
  %i32 = load i32, i32* %i, align 4
  %j33 = load i32, i32* %j, align 4
  %m234 = getelementptr [3 x [3 x i32]], [3 x [3 x i32]]* %m2, i32 0, i32 %i32, i32 %j33
  %m235 = load i32, i32* %m234, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 %m235)
  %j36 = load i32, i32* %j, align 4
  %tmp = add i32 %j36, 1
  store i32 %tmp, i32* %j, align 4
  br label %while30

merge:                                            ; preds = %while30
  %i39 = load i32, i32* %i, align 4
  %tmp40 = add i32 %i39, 1
  store i32 %tmp40, i32* %i, align 4
  br label %while

merge43:                                          ; preds = %while
  ret void
}
